### ISSUE TITLE
Flower (966330)

### ISSUE THREAD

**Issue by fsyy** [ValveSoftware/Proton#2483]
# Compatibility Report
- Name of the game with compatibility issues: Flower
- Steam AppID of the game: 966330

## System Information
- GPU: GTX 1060 6GB
- Driver/LLVM version: nvidia 418.56
- Kernel version: 5.0.5
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/fsyy/a9ad752924afbbb6e3655e827d6d9bca
- Proton version: 4.2-1 Beta

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

Log file: https://gist.github.com/fsyy/d5a2f6857e272068a4fd9d83c96c59e5

## Symptoms
if i start the game i get a black screen for a few seconds then it exits

## Reproduction
hit the play button

**Comment by kisak-valve** [comment]
These look like lines of interest:
```
fixme:d3dcompiler:D3DReadFileToBlob filename L"Data/Shaders/Release/SlideShow_vs.cso", contents 0x21c400
fixme:d3dcompiler:D3DReadFileToBlob filename L"Data/Shaders/Release/SlideShow_fs.cso", contents 0x21c400
```

**Comment by HonkingGoose** [comment]
I also see lines relating to ```mf.dll``` and  ```MFPlat.DLL``` in the Proton log, might this be Media Foundation related? Should this be added to my list at issue 1464?

**Comment by aqxa1** [comment]
winetricks d3dcompiler_47 (on Flower's pfx) will likely workaround the launch issue. You could try the [mfplat workaround](https://github.com/doitsujin/dxvk/issues/886#issuecomment-458487008), which might workaround further issues.

**Comment by fsyy** [comment]
yep with d3dcompiler_47 i see the starting screen, but no further.

i will try the mfplat workaround later. thanks for your suggestion.

**Comment by fsyy** [comment]
today proton 4.2-4 was released, it should fix the starting problem:

https://github.com/ValveSoftware/Proton/wiki/Changelog

i already tried it and i can tell it starts, i have sound, i see the flower logo on the top right, but as soon i move my mouse the logo flys away, i can't click anything.

i can enter the options menu if i press esc.

here is the latest log i created just now:

https://gist.github.com/fsyy/5820910b94817fd219ac7bb3ee5ffb3e

i'm using nvidia 418.74 on arch linux kernel 5.1.2

**Comment by AwesamLinux** [comment]
- GPU: RX VEGA 56
- Driver/LLVM version: Mesa 19.2.0-devel/9.0.0
- Distro version: Linux Mint 19.1 Tessa (64 bit) (Cinnamon Desktop)
- Proton version: 4.2-4

Game appears to work fine as of Proton 4.2-4 in fullscreen, in windowed mode it is unplayable.
- Nintendo Switch Pro Controller work
- Performance a solid 60fps with everything maxed at 1600p (not even any noticeable shader stuttering)

Log of playing through the first two stages in fullscreen for scrutiny: 
[steam-966330_gameplay.log](https://github.com/ValveSoftware/Proton/files/3183760/steam-966330_gameplay.log)
[system_info.txt](https://github.com/ValveSoftware/Proton/files/3183761/system_info.txt)

**Windowed Problem:**
Switching to windowed mode the game window is off-screen, and the window captures the cursor and it is impossible to move the window so that it is visible. If you try to drag the window it just jumps back.

**Reproduction:**
- Launch the game in fullscreen
- Switch to windowed
- Try to move the window

Captured a video of this problem, also log of this:
[steam-966330_problem.log](https://github.com/ValveSoftware/Proton/files/3183768/steam-966330_problem.log)
[steam-966330_problem.mp4.zip](https://github.com/ValveSoftware/Proton/files/3183769/steam-966330_problem.mp4.zip)

**Comment by AwesamLinux** [comment]
- GPU: RX VEGA 56
- Driver/LLVM version: 19.1.0+aco+git1564050652-58e0420cb35-1bionic1 (Valve-Experimental PPA)
- Kernel version: 5.2.3-050203-lowlatency
- Distro version: Linux Mint 19.1 Tessa (64 bit) (Cinnamon Desktop)
- Proton version: 4.2-9
- System information: [system_info.txt](https://github.com/ValveSoftware/Proton/files/3438303/system_info.txt)

I re-tested game with more recent Kernel/Proton/Mesa+ACO. The issue with the window behavior persists when switching between Fullscreen/Windowed.

[steam-966330.log.zip](https://github.com/ValveSoftware/Proton/files/3438304/steam-966330.log.zip)

**Comment by AwesamLinux** [comment]
- GPU: RX VEGA 56
- GPU Driver: Mesa ACO 19.1.0+aco+git1564433637-5c6077221dd-1bionic2 (Valves PPA)
- Kernel version:  5.2.5-050205-lowlatency
- Distro version: Linux Mint 19.1 Tessa (64 bit) (Cinnamon Desktop)
- Proton version: 4.11-1
- System information: [system_info.txt](https://github.com/ValveSoftware/Proton/files/3453373/system_info.txt)

Re-tested game and the game works without problems now. Switching between windowed and fullscreen mode works as expected. So this is might be whitelist material :tada: 

I have not played far enough to say for sure, if there is any problems left I would expect it to occur at the very ending cut-scene sequence.

Including a log of playing the game for a bit:
[steam-966330.log.zip](https://github.com/ValveSoftware/Proton/files/3453377/steam-966330.log.zip)

**Comment by mikaelhg** [comment]
I'm having an issue where the WASD keys don't control the petal at all. There is no way to move at all.

**Comment by natryamar** [comment]
Every other time this game is launched on steam deck the graphics are warped and stretched to the point where it is impossible to see what is going on. Once I restart the game the visuals go back to normal and vice versa after closing and launching it again. I have also experienced random crashes at random times.

The game experiences stuttering at any graphical setting until vertical sync is enabled and then disabled. Otherwise the "Auto detect graphics settings" option works perfectly. 

As far as I know it was possible to connect a dualshock 4 controller to your computer and this game would use it for the gyro motion controls like on the PS3 and PS4 versions of this game. Is there any way for the steam deck to take advantage of that feature for native gyro controls?

**Comment by waspennator** [comment]
Having a weird experience with this game on deck. Was gonna report an issue where the audio doesn't play save for cutscenes on the latest Experimental, but that just up and went away on further playthroughs. I'm also not noticing any weird graphical quirks that others mention from a noticable glance.

I am however having a game breaking issue where this game will crash every other start with this pop up. (Works fine launch one, crashes launch 2, fine on 3, etc.)

![download (2)](https://github.com/ValveSoftware/Proton/assets/78768499/8cd12e41-dd9a-45a5-b80d-2e15d154d9f6)
[steam-966330.log](https://github.com/ValveSoftware/Proton/files/11671303/steam-966330.log)

**Comment by ranplayer** [comment]
This game has an issue **just for the wayland environment** with Proton 10 and 11: it always reset your resolution when you relaunch it.

Proton logs: [steam-966330.tar.gz](https://github.com/user-attachments/files/31839568/steam-966330.tar.gz)

Specs:
- OS: Arch
- Kernel: 6.18.49
- GPU: RX 6700 XT (Mesa 26.1.8)
- CPU: Ryzen 7 5700G (8c/16t)
- DE: Plasma 6.7.4 (Wayland) -> no problem with X11
- Display device: 1440p 120hz
